### PR TITLE
Mitra/Refactor 404 page to TS

### PR DESCRIPTION
**Pull request by mitra-fs**
Changes:

-   https://redmine.deriv.cloud/issues/51680#Mitra_/_Refactor_-_Convert_error_404_page_to_TS

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/elements/query-image.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ const QueryImage = ({
     loading,
     width,
     ...props
-}: QueryImageProps): React.ReactNode => {
+}: QueryImageProps): JSX.Element => {
```
Comment (yashim-deriv) [inline src/components/elements/query-image.tsx:35]:
```suggestion
}: QueryImageProps): ReactElement => {
```
`ReactElement` is a more appropriate type to return for components with no children. ie; `<QueryImage />`

File: src/components/elements/query-image.tsx [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import React from 'react'
 import styled from 'styled-components'
-import { GatsbyImage, getImage } from 'gatsby-plugin-image'
+import { GatsbyImage, getImage, ImageDataLike } from 'gatsby-plugin-image'
 
 type QueryImageProps = {
     alt: string
     className?: string
-    data: Record<string, unknown>
+    data: ImageDataLike
```
Comment (sean-binary) [inline src/components/elements/query-image.tsx:8]: Great that you found this! 🎉 👍

Comment (prince-deriv) [reply src/components/elements/query-image.tsx:8]: @sean-binary  I remember suggesting this on your PR 😢  But it seems that you like @mitra-fs  more than me 🤣

Comment (sean-binary) [reply src/components/elements/query-image.tsx:8]: Really?? @prince-deriv I don't remember this 😂  My bad man!

Comment (mitra-fs) [reply src/components/elements/query-image.tsx:8]: the-simpsons-homer-simpson

**Comment by yashim-deriv** [comment]
@mitra-fs probably need to add an explicit return type on the 404 page. `ReactNode` or `ReactElement`

**Review by yashim-deriv — APPROVED** [review]
LGTM

**Pull request merged**